pkg/sql/logictest/tests/fakedist/fakedist_test: TestLogic_canary_stats failed

pkg/sql/logictest/tests/fakedist/fakedist_test.TestLogic_canary_stats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/20746440?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/20746440?buildTab=artifacts#/) on master @ [d90b48fdc3077318799afd2939837c79289156f4](https://github.com/cockroachdb/cockroach/commits/d90b48fdc3077318799afd2939837c79289156f4):


```
=== RUN   TestLogic_canary_stats
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/dff5f228bb64e0f91faef03cfd46f51f/logTestLogic_canary_stats447215288
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:142: cluster virtualization disabled in global scope due to issue: #156124 (expected label: C-bug)
[07:02:37] setting distsql_workmem='61631B';
[07:02:38] --- queries start here (file: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3570/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats)
[07:02:38] DROP TABLE IF EXISTS canary_table;;
[07:02:38] SET create_table_with_schema_locked=false;
[07:02:38] CREATE TABLE canary_table (x int primary key, y int, FAMILY (x, y)) WITH (sql_stats_canary_window = '20s');
INSERT INTO canary_table VALUES (1, 1);;
[07:02:38] SHOW CREATE TABLE canary_table;
[07:02:38] 	-- OK;
    logic.go:3244: let $table_creation_ts = 2025-11-18T07:02:38.317646Z
[07:02:38] ALTER TABLE canary_table SET (sql_stats_canary_window = '30s');;
[07:02:38] SHOW CREATE TABLE canary_table;
[07:02:38] 	-- OK;
[07:02:38] SELECT create_statement FROM crdb_internal.create_statements AS OF SYSTEM TIME '2025-11-18T07:02:38.317646Z' WHERE descriptor_name = 'canary_table';;
[07:02:42] 	-- OK;
[07:02:42] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3570/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats: 7 statements
[07:02:42] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3570/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats with config fakedist: 12 tests, 2 failures
[07:02:44] --- total progress: 12 statements
--- total: 12 tests, 2 failures
    logic.go:4721: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/dff5f228bb64e0f91faef03cfd46f51f/logTestLogic_canary_stats447215288
--- FAIL: TestLogic_canary_stats (9.48s)
=== RUN   TestLogic_canary_stats/canary_stats_with_query_cache
[07:02:42] SET CLUSTER SETTING sql.stats.canary_fraction='0';;
[07:02:42] SET TRACING = "on", cluster;;
[07:02:42] SELECT * FROM canary_table;;
[07:02:42] SET TRACING = "off";;
[07:02:42] SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';;
[07:02:42] 	-- FAIL
    logic.go:3194: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3570/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats:67: SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
        expected:
            query cache miss
            query cache add
            
        but found (query options: "") :
            query cache miss: SELECT * FROM canary_table
            query cache add: SELECT * FROM canary_table
            
[07:02:42] 	-- FAIL
    logic.go:2375: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3570/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats:73: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_canary_stats/canary_stats_with_query_cache (0.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_canary_stats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57044

pkg/sql/logictest/tests/fakedist/fakedist_test.TestLogic_canary_stats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/20752687?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/20752687?buildTab=artifacts#/) on master @ [4600cd303906f3d48bcb15c9f98db100a7b1a5a3](https://github.com/cockroachdb/cockroach/commits/4600cd303906f3d48bcb15c9f98db100a7b1a5a3):


```
             1048576  27            0            0            0            0            1      1044480            1      1044480
                     ---
Bytes in small extent cache: 0
HPA shard stats:
  Purge passes: 0 (0 / sec)
  Purges: 0 (0 / sec)
  Hugeifies: 0 (0 / sec)
  Dehugifies: 0 (0 / sec)

  In full slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 
  In empty slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 

                size ind npageslabs_huge    nactive_huge     ndirty_huge  npageslabs_nonhuge     nactive_nonhuge      ndirty_nonhuge   nretained_nonhuge
                     ---
--- End jemalloc statistics ---
[07:41:33] --- total progress: 12 statements
--- total: 12 tests, 2 failures
    logic.go:4721: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/dff5f228bb64e0f91faef03cfd46f51f/logTestLogic_canary_stats972916717
--- FAIL: TestLogic_canary_stats (11.70s)
=== RUN   TestLogic_canary_stats/canary_stats_with_query_cache
[07:41:31] SET CLUSTER SETTING sql.stats.canary_fraction='0';;
[07:41:31] SET TRACING = "on", cluster;;
[07:41:31] SELECT * FROM canary_table;;
[07:41:31] SET TRACING = "off";;
[07:41:31] SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';;
[07:41:31] 	-- FAIL
    logic.go:3194: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3659/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats:67: SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
        expected:
            query cache miss
            query cache add
            
        but found (query options: "") :
            query cache miss: SELECT * FROM canary_table
            query cache add: SELECT * FROM canary_table
            
[07:41:31] 	-- FAIL
    logic.go:2375: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3659/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats:73: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_canary_stats/canary_stats_with_query_cache (0.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_canary_stats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/logictest/tests/fakedist/fakedist_test.TestLogic_canary_stats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/20758742?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/20758742?buildTab=artifacts#/) on master @ [ef4e9e488e7889a4193d814150dc27646438c0b6](https://github.com/cockroachdb/cockroach/commits/ef4e9e488e7889a4193d814150dc27646438c0b6):


```
             1572864  29            0            0            0            0            1      1568768            1      1568768
                     ---
Bytes in small extent cache: 0
HPA shard stats:
  Purge passes: 0 (0 / sec)
  Purges: 0 (0 / sec)
  Hugeifies: 0 (0 / sec)
  Dehugifies: 0 (0 / sec)

  In full slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 
  In empty slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 

                size ind npageslabs_huge    nactive_huge     ndirty_huge  npageslabs_nonhuge     nactive_nonhuge      ndirty_nonhuge   nretained_nonhuge
                     ---
--- End jemalloc statistics ---
[07:22:22] --- total progress: 12 statements
--- total: 12 tests, 2 failures
    logic.go:4721: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/dff5f228bb64e0f91faef03cfd46f51f/logTestLogic_canary_stats858964629
--- FAIL: TestLogic_canary_stats (12.10s)
=== RUN   TestLogic_canary_stats/canary_stats_with_query_cache
[07:22:20] SET CLUSTER SETTING sql.stats.canary_fraction='0';;
[07:22:20] SET TRACING = "on", cluster;;
[07:22:20] SELECT * FROM canary_table;;
[07:22:20] SET TRACING = "off";;
[07:22:20] SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';;
[07:22:20] 	-- FAIL
    logic.go:3194: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3571/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats:67: SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
        expected:
            query cache miss
            query cache add
            
        but found (query options: "") :
            query cache miss: SELECT * FROM canary_table
            query cache add: SELECT * FROM canary_table
            
[07:22:20] 	-- FAIL
    logic.go:2375: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3571/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/canary_stats:73: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_canary_stats/canary_stats_with_query_cache (0.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_canary_stats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/testdata/logic_test/canary_stats
+++ b/pkg/sql/logictest/testdata/logic_test/canary_stats
@@ -64,7 +64,7 @@ SET TRACING = "off";
 
 # The first execution should miss the query cache.
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
 ----
 query cache miss
 query cache add
@@ -80,7 +80,7 @@ SET TRACING = "off";
 
 # The second execution should hit the query cache.
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
 ----
 query cache hit
 
@@ -99,7 +99,7 @@ statement ok
 SET TRACING = "off";
 
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
 ----
 not using query cache
 
@@ -113,7 +113,7 @@ statement ok
 SET TRACING = "off";
 
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
 ----
 not using query cache
 
@@ -133,7 +133,7 @@ SET TRACING = "off";
 
 # We should see the original cache is still there.
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%';
 ----
 query cache hit
 
@@ -188,7 +188,7 @@ statement ok
 SET TRACING = "off";
 
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
 ----
 query cache miss (prepare)
 query cache hit (prepare)
@@ -210,7 +210,7 @@ statement ok
 SET TRACING = "off";
 
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
 ----
 reusing cached memo
 reusing cached memo
@@ -243,7 +243,7 @@ statement ok
 SET TRACING = "off";
 
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
 ----
 query cache hit (prepare)
 query cache hit (prepare)
@@ -272,7 +272,7 @@ statement ok
 SET TRACING = "off";
 
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
 ----
 not using query cache
 not using query cache
@@ -304,7 +304,7 @@ statement ok
 SET TRACING = "off";
 
 query T
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
+SELECT split_part(message, ':', 1) FROM [SHOW TRACE FOR SESSION] WHERE message LIKE '%query cache%' OR message LIKE '%cached memo%';
 ----
 reusing cached memo
 reusing cached memo

